test.py

```diff
@@ -1338,29 +1338,67 @@ def format_unidiff(fromfile: str, tofile: str) -> str:
         return buf.getvalue()
 
 
+def update_junit_suite(suite, test):
```

Comment:
Missing a comment for the function. I know it's not customary here, but I comment nearly all functions I write, so you asked me for review.

Generally I think it's preferable to collect the data once and then create the element tree, not update the element tree on the fly.